agent/engine/dependencygraph/graph.go

```diff
@@ -183,38 +183,25 @@ func verifyTransitionDependenciesResolved(target *api.Container, existingContain
 		return true
 	}
 
-	for _, containerDependency := range target.TransitionDependencySet.ContainerDependencies {
-		maybeResolves, exists := existingContainers[containerDependency.ContainerName]
+	// TODO: add verifyResourceDependenciesResolved check here
+	return verifyContainerDependenciesResolved(target, existingContainers)
+}
+
+func verifyContainerDependenciesResolved(target *api.Container, existingContainers map[string]*api.Container) bool {
+	targetNext := target.GetKnownStatus() + 1
```

Comment:
A potential improvement here (not required at all in this PR, something to think about) is to instead get the next status from the "Container" object or some transition map. This will mean that containers/resources/tasks can potentially skip transitions if required as they wouldn't rely on the "+1" logic. Might be worth adding a TODO for.

Reply:
I think you're referring to:
https://github.com/aws/amazon-ecs-agent/blob/edc3e26016e363d7e8e6c044b11bde9bbb8853a8/agent/api/container.go#L367
Shall add in this PR.